gprofiler/utils.py

```diff
@@ -46,6 +46,7 @@
 hostname: Optional[str] = None
 
 
+@lru_cache(maxsize=None)
```

Comment:
use `functools.cache` if you are already passing None.

![image](https://user-images.githubusercontent.com/2903914/126326778-f9f47b21-b1ba-4d63-9db8-9f1fcc2d88ae.png)

Reply:
Too new (missing in e.g 3.7)